---
title: C6263
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C6263
helpviewer_keywords:
- C6263
ms.assetid: bc360ad7-5f59-4480-a642-6c7e6beeb5f6
ms.openlocfilehash: 8002a7da096c104f8c232b3b2f1d7ab55313a594
ms.sourcegitcommit: a1676bf6caae05ecd698f26ed80c08828722b237
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 09/29/2020
ms.locfileid: "91498896"
---
# <a name="c6263"></a>C6263

> AVERTISSEMENT C6263 : utilisation de _alloca dans une boucle ; Cela peut rapidement déborder la pile

Cet avertissement indique que l’appel de _alloca à l’intérieur d’une boucle pour allouer de la mémoire peut provoquer un dépassement de capacité de la pile. _alloca alloue de la mémoire à partir de la pile, mais cette mémoire est libérée uniquement lorsque la fonction appelante se termine. La pile, même en mode utilisateur, est limitée et l’échec de la validation d’une page de la pile provoque une exception de dépassement de capacité de la pile. La fonction `_resetstkoflw` récupère d’une condition de dépassement de capacité de la pile, ce qui permet à un programme de poursuivre son exécution au lieu d’échouer avec une erreur d’exception fatale. Si la `_resetstkoflw` fonction n’est pas appelée, aucune page de garde n’est présente après l’exception précédente. La prochaine fois qu’un dépassement de capacité de la pile se produit, aucune exception n’est déclarée et le processus s’achève sans avertissement.

Vous devez éviter `_alloca` d’appeler à l’intérieur d’une boucle si la taille d’allocation ou le nombre d’itérations est inconnu, car cela peut provoquer un dépassement de capacité de la pile. Dans ce cas, envisagez d’autres options, telles que, la mémoire du tas ou les classes de [bibliothèque standard C++](../standard-library/cpp-standard-library-reference.md) .

## <a name="example"></a>Exemple

Le code suivant génère cet avertissement :

```cpp
#include <windows.h>
#include <malloc.h>
#include <excpt.h>
#include <stdio.h>

#define MAX_SIZE 50

void f ( int size )
{
  char* cArray;
  __try
  {
    for(int i = 0; i < MAX_SIZE; i++)
    {
      cArray = (char *)_alloca(size);

     // process cArray...
    }
  }
  __except(GetExceptionCode() == STATUS_STACK_OVERFLOW ?
              EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH )
  {
     // code...
     puts("Allocation Failed");
    _resetstkoflw();
  }
}
```

Le code suivant utilise malloc () pour corriger cet avertissement :

```cpp
#include <windows.h>
#define MAX_SIZE 50

void f ( int size )
{
  char* cArray;

  for(int i = 0; i < MAX_SIZE; i++)
  {
     cArray = (char *) malloc(size);
     if (cArray != NULL)
     {
       // process cArray...
       free(cArray);
     }
  }
}
```

## <a name="see-also"></a>Voir aussi

- [malloc](../c-runtime-library/reference/malloc.md)
- [_alloca](../c-runtime-library/reference/alloca.md)
- [_malloca](../c-runtime-library/reference/malloca.md)
- [Bibliothèque standard C++](../standard-library/cpp-standard-library-reference.md)
